### PR TITLE
chore: Cluster sync should have short requeue, no backoff

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes # 

**Description**

- Right now, the cluster state syncing mechanism will have an exponential backoff which can create large gaps between runs of the provisioning or deprovisioning loop. This should not be an exponential backoff but a continued one second poll
- Also, the counter controller _should_ use cluster state sync when overwriting the status. Otherwise, there is a potential for us to overwrite the status with a smaller value and shoot over our limits

**How was this change tested?**

`make presubmit`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/controllers/provisioning/provisioner.go
Comment (bwagner5) [inline pkg/controllers/provisioning/provisioner.go:1]: does this change need to be coupled in this PR? Seems completely unrelated

File: pkg/controllers/deprovisioning/controller.go [hunk 1]
```diff
@@ -106,8 +106,11 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) controller.Bu
 }
 
 func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
+	// We need to ensure that our internal cluster state mechanism is synced before we proceed
+	// with making any scheduling decision off of our state nodes. Otherwise, we have the potential to make
+	// a scheduling decision based on a smaller subset of nodes in our cluster state than actually exist.
 	if !c.cluster.Synced(ctx) {
-		return reconcile.Result{Requeue: true}, nil
```
Comment (bwagner5) [inline pkg/controllers/deprovisioning/controller.go:110]: this was an immediate require, not an exponential backoff right?  An exponential backoff should only occur if an error is returned.

Comment (jonathan-innis) [reply pkg/controllers/deprovisioning/controller.go:110]: No, a requeue like this in controller-runtime is treated as a requeue with backoff. The only difference between this and the error version is that it won't print an error if you just do a `Requeue: true`

File: pkg/controllers/provisioning/provisioner.go
Comment (jonathan-innis) [reply pkg/controllers/provisioning/provisioner.go:1]: Sure, pulled it out

File: pkg/controllers/provisioning/provisioner.go [hunk 2]
```diff
@@ -114,8 +115,11 @@ func (p *Provisioner) Reconcile(ctx context.Context, _ reconcile.Request) (resul
 	if triggered := p.batcher.Wait(ctx); !triggered {
 		return reconcile.Result{}, nil
 	}
+	// We need to ensure that our internal cluster state mechanism is synced before we proceed
+	// with making any scheduling decision off of our state nodes. Otherwise, we have the potential to make
+	//	// a scheduling decision based on a smaller subset of nodes in our cluster state than actually exist.
```
Comment (bwagner5) [inline pkg/controllers/provisioning/provisioner.go:120]:
```suggestion
	// a scheduling decision based on a smaller subset of nodes in our cluster state than actually exist.
```
